### ISSUE TITLE
Cant Access Nova After Installation

### ISSUE THREAD

**Issue by zek** [lorisleiva/laravel-actions#7]
After I install this package I can no longer access Laravel Nova. Just returns 404 with no error message.

But everything works fine when I remove library.

**Comment by lorisleiva** [comment]
Hi there 👋 

Thanks for reporting this. Could you provide more information on the error itself? Are the routes from Nova not loaded anymore or is it an issue within the Nova ecosystem?

**Comment by zek** [comment]
I actually don't know I only installed library then I notice when I try to access nova I got 404. 

There is no error logs :)

**Comment by zek** [comment]
~~~
        $this->app->extend('events', function ($dispatcher, $app) {
            return (new EventDispatcher($app))->setQueueResolver(function () use ($app) {
                return $app->make(QueueFactoryContract::class);
            });
        });
~~~
BTW when I remove this, everything works

**Comment by lorisleiva** [comment]
Thanks that’s very useful. Are you using the latest release? If you are, can you move that from the `boot` method to a new `register` method on that provider. I've recently moved this to the boot method and I'm wondering if it's causing some since effects.

**Comment by zek** [comment]
@lorisleiva Yeah it works after moving to register :)

**Comment by lorisleiva** [comment]
Okay so basically it doesn't work for certain users in the `boot` method and does not work for others in the `register` method haha. I'm looking into an alternative way to override the dispatcher. I'm thinking of decorating it instead of overriding it. It's difficult for me to debug this issue without having access to the Laravel Nova service providers. Could you maybe provide this to me?

**Comment by lorisleiva** [comment]
I've managed to access the code of some open source applications that use Laravel Nova. I've tried a few of them after installing Laravel Actions and I'm afraid I simply cannot reproduce this issue.

I'm going to need more information from this error in order to reproduce it and then fix it. 🙏 

I've also created a new branch called [`lab/decorate_event_dispatcher`](https://github.com/lorisleiva/laravel-actions/tree/lab/decorate_event_dispatcher) that changes the way Laravel Actions extends the event dispatcher by decorating it.

```php
$this->app->extend('events', function ($dispatcher, $app) {
    return new EventDispatcherDecorator($dispatcher, $app);
});
```

Could you give this branch a try and tell me if it changes anything? You can upgrade to this branch by running:

```sh
composer require lorisleiva/laravel-actions:dev-lab/decorate_event_dispatcher
```

**Comment by zek** [comment]
It works like a charm 👍 

Thank you so much for this awesome library and all your effort solving this issue.

Also I admired all of your work. You got a follower :) I Sent a message on twitter

**Comment by lorisleiva** [comment]
Thanks, I'll release that properly very soon! 👍

**Comment by cbaconnier** [comment]
I was about to create an issue about that, I just installed it and it broke all my listeners (I don't use Nova btw so I think it was more general). 
I'm glad you solved it 🥳

One weird issue the fix added: `Carbon` uses the current localization, so, when you do `today()->monthName` and your application is in french, you expect to get the month in french. 
It seems your fix `lab/decorate_event_dispatcher` broke that for me and returns it in english. I tried to check where Laravel was handling the locale for Carbon but I did not succeed.
Using Laravel 5.8.19

Forcing `Carbon::setLocale(config('app.locale'));` on my AppServiceProvider solved it though. 

_(The bug does not appear on 0.1.2 and IRCC, the automatic localization came as improvement with laravel 5.8 and the venue of Carbon 2.0)_

Excellent work on this package it was exactly what I was looking for!

**Comment by cbaconnier** [comment]
About my the new issue that has been added with `lab/decorate_event_dispatcher`
here's the root of the problem: 
https://github.com/briannesbitt/Carbon/blob/master/src/Carbon/Laravel/ServiceProvider.php#L22

Carbon expect to have `Illuminate\Events\Dispatcher` but get `Illuminate\Contracts\Events\Dispatcher`

**Comment by lorisleiva** [comment]
Awesome! Thanks for your detailed report. I’ll make the necessary changes to fix that in the release. Btw, I’m thinking of jumping from `0.1.2` to `0.2.0` but I’m not sure that this release actually contains a breaking change. What do you think?

**Comment by cbaconnier** [comment]
To be honest, I'm not entirely sure to be enough experimented to give you a good answer. From my point of view, you're in version `0.^` so, for my understanding it's still an "alpha" stage and all releases until 1.0 may get several breaking change. So it's more like a "use it as your own risk".

However, the current state of `0.1.2` seems to be broke for Laravel 5.8 while `0.2.0` would fix it adding a small breaking change with Carbon (but easily fixable by the users if necessary). So I think it's a good trade off if this tag is still compatible with Laravel 5.7

**Comment by lorisleiva** [comment]
Thanks for your feedback @cbaconnier!

Regarding the Carbon bug, I've created a [new issue](https://github.com/briannesbitt/Carbon/issues/1747) on their repository to ask them to check the event dispatcher using its contract.

If this isn't happening it means we're going to have to extend the implementation of the current event dispatcher and it's a lot more difficult to decorate and maintain as all of its methods will have to be overridden. It will be more of a hacky decorator than a proper decorator.

Therefore, I'm hoping we can fix this issue on their side (as it makes absolute sense in the OOP world and, otherwise, I don't see the point of Laravel using contracts). I'm also hoping that we're not going to have more side-effects like this from other libraries.

**Comment by lorisleiva** [comment]
A PR was [welcomed](https://github.com/briannesbitt/Carbon/issues/1747#issuecomment-498568328) by the maintainer and is now [created](https://github.com/briannesbitt/Carbon/pull/1748). 🤞

**Comment by lorisleiva** [comment]
The PR has been merged! 🎉 

@cbaconnier Could you please let me know if using the `lab` branch with `carbon:dev-master` fixes everything for you?

**Comment by cbaconnier** [comment]
I couldn't update directly Carbon to 2.19.1 since it's in the Laravel dependencies. I had to copy manually the files but all my tests went back to green once I did it, so I guess it's working! :tada: 

Thank you very much for all your efforts and your work @lorisleiva :bowing_man:

**Comment by exodusanto** [comment]
Is there a release in schedule about this issue? 👨🏻‍💻

**Comment by lorisleiva** [comment]
The fix is included in the new `v0.1.3` release.
